### ISSUE TITLE
The  priority level of options when they are given in input-file, commind-line, and aria2.conf file simultaneously.

### ISSUE THREAD

**Issue by hongyi-zhao** [tatsuhiro-t/aria2#424]
Hi Tatsuhiro,

The options of aria2c can be given in the following locations: input-file, commind-line, and aria2.conf file.  But the issue is: when I given some of them in the input-file, commind-line, and aria2.conf file simultaneously and the same option has diffent values in these locations, then what's the priority level of these options, i.e., which value will take effect finally for the corresponding options?

Regards

**Comment by tatsuhiro-t** [comment]
In general, input-file, command-line, aria2.conf, in the order of precedence.

**Comment by hongyi-zhao** [comment]
What about the rules using by curl: 

If a option is used several times, the last one will be used.

Regards

**Comment by tatsuhiro-t** [comment]
In general, last one overwrites previous one, in the same way curl does.
Several options are treated differently, and it gathers all given parameters and treats them as list (e.g., --header)

**Comment by tatsuhiro-t** [comment]
... or "concatenated" in the option specific way.

**Comment by hongyi-zhao** [comment]
I've tried the --daemon option as follows:

Set it in the aria2.conf file, and at the same time, using a different value in the command-line for this option.

And, I found that the setting in the command-line will always overwrite the one set in the aria2.conf file regardless of the order of the appearance of the them in the invoking sequece for this option.

I mean, this behaviour is violate with your above depictions: 
  
In general, last one overwrites previous one, in the same way curl does.

Regards
